### PR TITLE
[code-coverage] Add more tests for service/history/decision package 

### PR DESCRIPTION
**Pull request by ketsiambaku**
**What changed?**
- Generated mock for service/history/query/registry interface
- Cover additional scenarios in HandleBufferedQueries API
- Added tests for CreateRecordDecisionTaskStartedResponse API
- Added more tests for HandleDecisionTaskStarted API

**Why?**
improve code coverage in service/history/decision

**How did you test it?**
unit tests

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/decision/handler_test.go [hunk 1]
```diff
@@ -462,11 +472,43 @@ func (s *DecisionHandlerSuite) TestHandleDecisionTaskStarted() {
...
+				engine.EXPECT().NotifyNewReplicationTasks(gomock.Any())
+			},
+			expectErr: nil,
+			mutablestate: &persistence.WorkflowMutableState{
+				ExecutionInfo: &persistence.WorkflowExecutionInfo{
+					DecisionStartedID: -23,
+					NextEventID:       2,
+				},
+			},
+			assertCalls: func(resp *types.RecordDecisionTaskStartedResponse) {},
```
Comment (shijiesheng) [inline service/history/decision/handler_test.go:511]: Should we assert the response body like the previous success?

**Review by shijiesheng — APPROVED** [review]

File: service/history/decision/handler_test.go [hunk 2]
```diff
@@ -547,6 +653,84 @@ func (s *DecisionHandlerSuite) TestHandleBufferedQueries_QueryTooLarge() {
...
+			queryResults:         map[string]*types.WorkflowQueryResult{"some-id": &types.WorkflowQueryResult{}},
+			assertCalls: func(logs *observer.ObservedLogs) {
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to completed").Len())
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to unblocked").Len())
+			},
+		},
+	}
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			core, observedLogs := observer.New(zap.ErrorLevel)
```
Comment (davidporter-id-au) [inline service/history/decision/handler_test.go:723]: oh nice! I didn't know about this

File: service/history/decision/handler_test.go [hunk 3]
```diff
@@ -547,6 +653,84 @@ func (s *DecisionHandlerSuite) TestHandleBufferedQueries_QueryTooLarge() {
...
+			clientFeatureVersion: client.GoWorkerConsistentQueryVersion,
+			queryResults:         map[string]*types.WorkflowQueryResult{"some-id": &types.WorkflowQueryResult{}},
+			assertCalls: func(logs *observer.ObservedLogs) {
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to completed").Len())
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to unblocked").Len())
+			},
+		},
+	}
+	for _, test := range tests {
+		s.Run(test.name, func() {
```
Comment (davidporter-id-au) [inline service/history/decision/handler_test.go:722]: you're kinda using a table test and a suite test at the same time which - normally I'd say is probably a bit confusing, but I take it that there's existing tests in here that you're re-using. One thing that *might* cause problems is that the `s.` may maintain state between tests, (like on line 725) so I'd suggest if possible moving to just use plain `t.Run()` and creating a new decisionHandler every time in future.

**Review by davidporter-id-au — APPROVED** [review]

File: service/history/decision/handler_test.go (hunk 3 above)
Comment (ketsiambaku) [reply service/history/decision/handler_test.go:722]: changed in #5932

File: service/history/decision/handler_test.go (hunk 1 above)
Comment (ketsiambaku) [reply service/history/decision/handler_test.go:511]: Yes, thank you! Added in #5932

**Pull request merged**